src/libraries/System.Private.CoreLib/src/System/Resources/ResourceSet.cs

```diff
@@ -208,13 +185,15 @@ private IDictionaryEnumerator GetEnumeratorHelper()
 
         protected virtual void ReadResources()
         {
-            Debug.Assert(Table != null);
+            Debug.Assert(_table != null);
             Debug.Assert(Reader != null);
             IDictionaryEnumerator en = Reader.GetEnumerator();
             while (en.MoveNext())
             {
-                object? value = en.Value;
-                Table.Add(en.Key, value);
+                if (en.Key is not string key)
```

Comment:
Should we assert here?

Reply:
It can happen so probably not unless you want to replace the if with assert

Reply:
If it can happen, then aren't we changing behavior here?

If the ResourceReader returned a non-string key, then `ResourceSet.GetEnumerator` will no longer return it.

Reply:
By it can happen I meant it's a weird side effect of using weakly types collection. You are right that `ResourceSet.GetEnumerator` will not return now. Is that something what we should keep when the whole API is string-based?

Reply:
@ericstj @tarekgh @JeremyKuhne - do any of you know arguments for/against keeping support for non-string keyed resources?

Reply:
I am not aware of any scenario that need to create resources with non string keys. 

Also, if you look at the ResourceReader, the enumerator is always returning the keys as strings https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs#L1025. 

But, still unknown to me if there will be anyone ever can be broken by this. is there anything prevent us to make the dictionary of object keys instead of strings? just to be in the safe side.

Reply:
I think that could work too

Reply:
Changed to Dictionary<object, object?>